fixql-core/src/main/scala/com/iterable/graphql/Document.scala

```diff
@@ -0,0 +1,50 @@
+package com.iterable.graphql
+
+import cats.Functor
+import play.api.libs.json.JsObject
+import play.api.libs.json.Json
+import qq.droste.data.Attr
+import qq.droste.data.Fix
+
+/** Simplified AST for "reduced" documents: variables have been interpolated,
+  * and fragments have been inlined
+  *
+  * This is unused today but may be required when we introduce support for mutations.
+  */
+case class Document[T](
+  query: Query[T]
+)
+
+case class Query[A](
+  fields: Seq[A]
+) {
+  // TODO: move to sealed trait or let Query == Root
+  def fieldTreeRoot = Field("", subfields = fields)
+
+  def map[B](f: A => B) = {
+    Query(fields.map(f))
+  }
+}
+
+/**
+  * @param arguments we use a JsObject because we may want to support case class deserialization here
+  *                  and JSON codecs capture the reality of what's being passed
+  */
+case class Field[A](
+  name: String,
+  arguments: JsObject = Json.obj(),
+  subfields: Seq[A] = Nil,
+) {
+  def map[B](f: A => B): Field[B] = {
+    Field(name, arguments, subfields.map(f))
+  }
+}
```

Comment:
The case classes above encode a very simple AST for GraphQL documents (queries).

Reply:
We use a GADT to make it work with the fixpoint data types.